lesson_30_allure/tests/test_user_registration.py

```diff
@@ -0,0 +1,143 @@
+import allure
+import pytest
+import sys
+import os
+from selenium.webdriver.common.by import By
+from lesson_30_allure.tests.conftest import register_user
+from lesson_30_allure.tests.conftest import navigate_to_site, driver
+
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), "../..")))
+
+from lesson_30_allure.pages.main_page import MainPage
+
+
+@allure.feature("Registration")
+@pytest.mark.usefixtures("navigate_to_site")
+def test_successful_registration(driver, register_user):
+    main_page = MainPage(driver)
+
+    main_page.click_signup_button()
+
+    # waiting for the registration form to appear.
+    main_page.wait_for_element_visibility((By.CLASS_NAME, "modal-body"))
+
+    main_page.fill_form("Test", "User", "k7muser@example.com", "ValidPass123", "ValidPass123")
+
+    # submitting the form
+    main_page.submit_form()
+
+    main_page.wait_for_registration_to_complete()
+
+    # checking the redirect to the profile
+    main_page.check_redirect_to_profile()
+
+
+@allure.feature("Registration")
+@pytest.mark.usefixtures("navigate_to_site")
+def test_existing_user_registration(driver, register_user):
+    main_page = MainPage(driver)
+
+    main_page.click_signup_button()
+
+    main_page.wait_for_element_visibility((By.CLASS_NAME, "modal-body"))
+
+    main_page.fill_form("Test", "User", "existing@example.com", "ValidPass123", "ValidPass123")
+
+    main_page.submit_form()
+
+    error_selector = (By.CLASS_NAME, "alert-danger")
+    error_message = main_page.wait_for_error_message(error_selector)
+
+    assert error_message == "User already exists", (
+        f"Expected error message: 'User already exists', but received: '{error_message}'"
+    )
+
+
+@allure.feature("Registration")
+@pytest.mark.usefixtures("navigate_to_site")
+def test_invalid_name_registration(driver):
+    main_page = MainPage(driver)
+
+    main_page.click_signup_button()
+
+    main_page.wait_for_element_visibility((By.CLASS_NAME, "modal-body"))
+
+    main_page.fill_field((By.ID, "signupName"), "")
+
+    driver.find_element(By.TAG_NAME, "body").click()
+
+    error_selector = (By.CSS_SELECTOR, "#signupName + .invalid-feedback")
```

Comment:
test should not contain selectors like (By.CLASS_NAME, "modal-body")...they should be hidden on page objects